api/src/main/java/org/openmrs/module/attachments/obs/Attachment.java

```diff
@@ -46,6 +52,9 @@ public Attachment(Obs obs) {
 		setDateTime(obs.getObsDatetime());
 		setComment(obs.getComment());
 		setComplexData(obs.getComplexData());
+		setBytesMimeType(complexDataHelper.getContentType(obs.getComplexData()));
```

Comment:
This line is trying to get the complex data of an obs that may not necessarily be a complex obs. In cases where this is not a complex obs, `Obs.getComplexData()` returns `Null`, hence the `NullPointerException`. The code that actually throws that exception is

```java
@Override
	public String getContentType(ComplexData complexData) {
		if (StringUtils.isEmpty(complexData.getMimeType())) {
			return super.getContentType(complexData);
		} else {
			return complexData.getMimeType();
		}
	}
```
`complexData` is null so `complexData.getMimeType` throws a `NullPointerException`

Reply:
@ivange94  @mks-d kindly review,,,

sorry about the revert ,wanted to create a clean pr because the previous one was ugly